### PR TITLE
Tests are done

### PR DESCRIPTION
**Pull request by Csillag61**

### CODE REVIEW

File: basic-javascript/project-1.spec.js [hunk 1]
```diff
@@ -1,3 +1,129 @@
...
+
+it("should add two numbers together.", () => {
+    expect(helpers.add(8,5)).toBe(13);
+});
+it('should subtract two numbers', () => {
+    expect(helpers.subtract(5,5)).toBe(0);
+});
+it('should divide two numbers', () => {
+    expect(helpers.divide(10,5)).toBe(2);
+});
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:52]: What if the denominator is zero?

File: basic-javascript/project-1.spec.js [hunk 2]
```diff
@@ -1,3 +1,129 @@
...
+    expect(helpers.greaterThanFifty(55)).toBe(true);
+    expect(helpers.greaterThanFifty(45)).toBe(false);
+    expect(helpers.greaterThanFifty('55')).toBe(false);
+    expect(helpers.greaterThanFifty(50)).toBe(false);
+
+});
+
+it("should add two numbers together.", () => {
+    expect(helpers.add(8,5)).toBe(13);
+});
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:46]: What if an input is a string?

File: basic-javascript/project-1.spec.js [hunk 3]
```diff
@@ -1,3 +1,129 @@
...
+    it('should round up a number', () => {
+        expect(helpers.roundUp(2.01)).toBe(3);
+        expect(helpers.roundUp(2.99)).toBe(3); 
+    });
+    
+    it('should add an exclamation point to the end of a string', () => {
+        expect(helpers.addExclamationPoint('Kitty')).toBe('Kitty!');
+        expect(helpers.addExclamationPoint('')).toBe('!'); 
+  
+    });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:100]: What if a number is supplied?

File: basic-javascript/project-1.spec.js [hunk 4]
```diff
@@ -1,3 +1,129 @@
...
+    
+    it('should combine two strings into one string', () => {
+        expect(helpers.combineNames('Csilla', 'Gates')).toBe('Csilla Gates')
+    });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:104]: What if one of the arguments is not supplied?

File: basic-javascript/project-2.spec.js [hunk 5]
```diff
@@ -1,3 +1,120 @@
 const funcs = require('./project-2');
 
 // whoops.. there is no test suite for this file. You'll simply just have to create one :/
+it('return the larger of the two numbers', () => {
+   expect(funcs.getBiggest(9, 8)).toBe(9);
+    expect(funcs.getBiggest(10, 10)).toBe(10);
+    expect(funcs.getBiggest(9, 10)).toBe(10);
+   
+});
```
Comment (gooseandmegander) [inline basic-javascript/project-2.spec.js:9]: What if a string is supplied?

File: basic-javascript/project-2.spec.js [hunk 6]
```diff
@@ -1,3 +1,120 @@
...
+ expect(funcs.isPrime(-1)).toBe(false);
+   expect(funcs.isPrime(1)).toBe(false);
+ expect(funcs.isPrime(2)).toBe(true);
+}); 
+
+it('should return first item of an array', () => {
+    expect(funcs.returnFirst([1, 2, 3])).toBe(1);
+   expect(funcs.returnFirst(['string', 1])).toBe('string');
+  expect(funcs.returnFirst(['str1', 'str2', true])).toBe('str1'); 
+});
```
Comment (gooseandmegander) [inline basic-javascript/project-2.spec.js:57]: What if an array is not supplied or another data type is supplied instead?

**Review by gooseandmegander — COMMENTED** [review]
# Objectives
* Familiarity with writing unit tests for JavaScript code.

# MVP
- [x] project-1.spec.js tests passing
- [x] project-2.spec.js tests passing

# Stretch
- [x] arrays.spec.js tests passing
- [x] objects.spec.js tests passing

# Great
All the tests pass. There are "green" and "red" tests as well as many edge cases accounted for. Very thorough!

# Requests for Improvement
* Check out other types of testing or even look into testing environments.
* Use something new from the docs you haven't used yet to generate tests.
* _Why would you want to keep your tests even though what you're testing is passing those tests?_

# Rating: 3
